### PR TITLE
#137156979 Implement notification system 

### PR DESCRIPTION
**Pull request by andela-oakinseye**
#### What does this PR do?
Implement notification system

#### Description of Task to be completed?
Add a notification system that ensures that users and vendors are notified of order activities.

#### How should this be manually tested?
Purchase a product and get the message, always watch to see your order status changing.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#137156979 Notifications

#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: .meteor/packages [hunk 1]
```diff
@@ -23,7 +23,7 @@ random@1.0.10
 reactive-var@1.0.11
 reactive-dict@1.1.8
 check@1.2.4
-http@1.2.10
+http
```
Comment (adebayo-maborukoje) [inline .meteor/packages]: let's maintain a version on this packages.

File: server/methods/core/orders.js [hunk 2]
```diff
@@ -324,6 +324,83 @@ Meteor.methods({
...
+            Logger.info("SMS SENT");
+          }
+        });
+      });
+    } else if (order.workflow.status === "coreOrderWorkflow/processing") {
+      smsContent.message = "Your orders is on the way and will soon be delivered.";
+      Meteor.call("send/smsAlert", smsContent, (error) => {
+        if (error) {
+          Logger.warn("ERROR", error);
+        } else {
```
Comment (adebayo-maborukoje) [inline server/methods/core/orders.js]: is this else statement necessary here?

File: server/methods/core/orders.js [hunk 3]
```diff
@@ -436,12 +513,34 @@ Meteor.methods({
...
   },
 
+  "send/smsAlert": function (smsContent) {
+    check(smsContent, Object);
+    HTTP.call("GET", "http://www.smslive247.com/http/index.aspx",
+      {
+        params:
+        {
+          cmd: "sendquickmsg",
+          owneremail: "pleroonigeria@gmail.com",
```
Comment (adebayo-maborukoje) [inline server/methods/core/orders.js]: whose email is this one? and why is it hardcoded?

File: server/methods/core/orders.js (hunk 2 above)
Comment (brainyfarm) [reply server/methods/core/orders.js]: Fixed.

File: server/methods/core/orders.js [hunk 4]
```diff
@@ -324,6 +324,83 @@ Meteor.methods({
...
+    const shoppersPhone = order.billing[0].address.phone;
+
+    //  Loop through orders to get vendor information and send notification to them
+    const orderedItems = order.items;
+    let orderedProducts = "";
+    let vendorPhones = [];
+    for (let i = 0; i < orderedItems.length; i += 1) {
+      orderedProducts += ` ${orderedItems[i].title},`;
+      const productVendor = Accounts.find({ _id: orderedItems[i].vendorId }).fetch();
+      Logger.info("Vendor Object", productVendor);
```
Comment (adebayo-maborukoje) [inline server/methods/core/orders.js]: Any reason why you are logging all this guys to the console?

File: server/methods/core/orders.js [hunk 5]
```diff
@@ -324,6 +324,83 @@ Meteor.methods({
...
+      vendorPhones.push(productVendor[0].profile.vendorDetails.vendorPhone);
+    }
+    Logger.info("Vendor Phones: ", vendorPhones);
+
```
Comment (adebayo-maborukoje) [inline server/methods/core/orders.js]: reduce the space here

File: server/methods/core/orders.js [hunk 6]
```diff
@@ -324,6 +324,83 @@ Meteor.methods({
...
+      Meteor.call("send/smsAlert", smsContent, (error) => {
+        if (error) {
+          Logger.warn("ERROR", error);
+        } else {
+          Logger.info("SMS SENT");
+        }
+      });
+    } else if (order.workflow.status === "coreOrderWorkflow/completed") {
+      smsContent.message = "Your orders on Banasko-RC has been shipped, thanks";
+      Meteor.call("send/smsAlert", smsContent, (error) => {
```
Comment (adebayo-maborukoje) [inline server/methods/core/orders.js]: since this method is repeated in other parts of the code, could you abstract it and reuse? keep it DRY

**Review by adebayo-maborukoje — APPROVED** [review]

**Pull request merged**